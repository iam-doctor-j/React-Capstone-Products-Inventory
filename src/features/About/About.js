import React from 'react';
import Card from 'react-bootstrap/Card';
import { refresh } from '../Authentication/AuthenticationSlice';
import { connect } from 'react-redux';

export const About = (props) => {
    return(
        <>
            <div className="container">
                <div className="row justify-content-center">
                <Card className="col-12 col-md-8 col-lg-6">
                    <Card.Body>
                        <Card.Title>
                            <div className="d-flex">
                                <span className="heading-text">About</span>
                                <a className="ml-auto" onClick={() => {props.dispatch(refresh()); props.history.goBack()}} style={{cursor: 'pointer', color: 'var(--secondary)'}}><i className="fas fa-arrow-circle-left"></i></a>
                            </div>
                        </Card.Title>
                        <div>
                            <p>Get a list of products, created by you or by others.</p>
                            <p>Add your own product, manage, delete and a lot of stuff.</p>
                        </div>
                    </Card.Body>
                </Card>
                </div>
            </div>
        </>
    );
}

export default connect()(About);