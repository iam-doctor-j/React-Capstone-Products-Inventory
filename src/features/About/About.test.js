import React from 'react';
import { About } from './About';
import { shallow } from 'enzyme';

describe('Test About with shallow rendering', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<About/>);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should display heading as About', () => {
        expect(wrapper.find('span.heading-text').render().text()).toEqual('About');
    })

    afterEach(() => {
        wrapper.unmount();
    })
})