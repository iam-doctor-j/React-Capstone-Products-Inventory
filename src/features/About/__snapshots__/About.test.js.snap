// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test About with shallow rendering should render correctly 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="row justify-content-center"
    >
      <Card
        body={false}
        className="col-12 col-md-8 col-lg-6"
      >
        <CardBody>
          <CardTitle>
            <div
              className="d-flex"
            >
              <span
                className="heading-text"
              >
                About
              </span>
              <a
                className="ml-auto"
                onClick={[Function]}
                style={
                  Object {
                    "color": "var(--secondary)",
                    "cursor": "pointer",
                  }
                }
              >
                <i
                  className="fas fa-arrow-circle-left"
                />
              </a>
            </div>
          </CardTitle>
          <div>
            <p>
              Get a list of products, created by you or by others.
            </p>
            <p>
              Add your own product, manage, delete and a lot of stuff.
            </p>
          </div>
        </CardBody>
      </Card>
    </div>
  </div>
</Fragment>
`;
